src/app/clusters/scenes/SceneTable.h

```diff
@@ -0,0 +1,244 @@
+            return (this->sceneEndpointId == other.sceneEndpointId && this->sceneGroupId == other.sceneGroupId &&
+                    this->sceneId == other.sceneId);
+        }
+        void operator=(const SceneStorageId & other)
+        {
+            this->sceneEndpointId = other.sceneEndpointId;
+            this->sceneGroupId    = other.sceneGroupId;
+            this->sceneId         = other.sceneId;
+        }
+    };
+
+    /// @brief struct used to store data held in a scene
+    struct SceneData
+    {
+        static constexpr TLV::Tag TagSceneName() { return TLV::ContextTag(1); }
+        static constexpr TLV::Tag TagSceneTransitionTime() { return TLV::ContextTag(2); }
+        static constexpr TLV::Tag TagSceneTransitionTime100() { return TLV::ContextTag(3); }
+
+        char name[kSceneNameMax]                = { 0 };
+        SceneTransitionTime sceneTransitionTime = 0;
+        ExtensionFieldsSets extentsionFieldsSets;
+        TransitionTime100ms transitionTime100 = 0;
+
+        SceneData() = default;
+        SceneData(const char * sceneName, SceneTransitionTime time = 0, TransitionTime100ms time100ms = 0) :
+            sceneTransitionTime(time), transitionTime100(time100ms)
+        {
+            SetName(sceneName);
+        }
+        SceneData(const char * sceneName, ExtensionFieldsSets fields, SceneTransitionTime time = 0,
+                  TransitionTime100ms time100ms = 0) :
+            sceneTransitionTime(time),
+            transitionTime100(time100ms)
+        {
+            SetName(sceneName);
+            extentsionFieldsSets = fields;
+        }
+        SceneData(const SceneData & data) : sceneTransitionTime(data.sceneTransitionTime), transitionTime100(data.transitionTime100)
+        {
+            SetName(data.name);
+            extentsionFieldsSets = data.extentsionFieldsSets;
+        }
+
+        CHIP_ERROR Serialize(TLV::TLVWriter & writer) const
+        {
+            TLV::TLVType container;
+            ReturnErrorOnFailure(writer.StartContainer(TLV::ContextTag(1), TLV::kTLVType_Structure, container));
+
+            size_t name_size = strnlen(this->name, kSceneNameMax);
+            ReturnErrorOnFailure(writer.PutString(TagSceneName(), this->name, static_cast<uint32_t>(name_size)));
+            ReturnErrorOnFailure(writer.Put(TagSceneTransitionTime(), static_cast<uint16_t>(this->sceneTransitionTime)));
+            ReturnErrorOnFailure(writer.Put(TagSceneTransitionTime100(), static_cast<uint8_t>(this->transitionTime100)));
+            ReturnErrorOnFailure(this->extentsionFieldsSets.Serialize(writer));
+
+            return writer.EndContainer(container);
+        }
+        CHIP_ERROR Deserialize(TLV::TLVReader & reader)
+        {
+            TLV::TLVType container;
+            ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::ContextTag(1)));
+            ReturnErrorOnFailure(reader.EnterContainer(container));
+
+            ReturnErrorOnFailure(reader.Next(TagSceneName()));

```

Comment:
You should not rely on the name always being provided. Please make the code resilient to missing name, and only put the name into storage if there is actually a name to store.